### PR TITLE
adjust logic for handling submit

### PR DESCRIPTION
**Pull request by jakekidd**

### CODE REVIEW

File: modules/contracts/src.ts/services/ethService.ts [hunk 1]
```diff
@@ -368,53 +373,72 @@ export class EthereumChainService extends EthereumChainReader implements IVector
...
           } else {
-            response = tempResponse;
+            throw error;
           }
-
-          // NOTE: Response MUST be defined here because if it was NEVER defined (i.e. undefined on first iteration),
-          // we would have returned in prev block, and if it was undefined on this iteration we would not overwrite
-          // that value.
-          // Tx was submitted: handle saving to store.
-          await this.handleTxSubmit(onchainTransactionId, method, methodId, channelAddress, reason, response!);
```
Comment (jakekidd) [inline modules/contracts/src.ts/services/ethService.ts:407]: So, working theory is that this line and the `response = tempResponse` line above were the problem lines.

Somehow, there was a response of `undefined` in a **subsuquent ** (2nd or greater number attempt) and we were *not* saving `tempResponse` to `response`... so we were calling `this.handleTxSubmit` and passing in a **stale response from the previous run**.

This resulted in uncaught prisma error - the txHash must be unique

Comment (jakekidd) [reply modules/contracts/src.ts/services/ethService.ts:407]: As to why we got `undefined` in a subsequent response, I don't know. @LayneHaber or @rhlsthrm maybe you can help me out with that. @LayneHaber I think there was some reason for it... maybe re-explain. This might be worth writing a unit test for that edge case

File: modules/contracts/src.ts/services/ethService.ts [hunk 2]
```diff
@@ -368,53 +373,77 @@ export class EthereumChainService extends EthereumChainReader implements IVector
...
-          // passed into txFn to bail out at the time of sending.
-          let tempResponse = result.getValue();
-          if (!tempResponse) {
-            this.log.warn({ method, methodId, channelAddress, reason }, "Did not attempt tx");
-            // Iff this is the only iteration, then we want to go ahead return w/o saving anything.
-            if (tryNumber === 1) {
-              return Result.ok(undefined);
+            // We must check for confirmation in all previous transactions. Although it's most likely
+            // that it's the previous one, any of them could have been confirmed.
+            for (let i = 0; i < responses.length; i++) {
```
Comment (jakekidd) [inline modules/contracts/src.ts/services/ethService.ts:428]: iter backwards

File: modules/contracts/src.ts/services/ethService.ts [hunk 3]
```diff
@@ -368,53 +373,77 @@ export class EthereumChainService extends EthereumChainReader implements IVector
...
+              try {
+                receipt = await this.waitForConfirmation(chainId, responses[i]);
+              } catch {}
```
Comment (jakekidd) [inline modules/contracts/src.ts/services/ethService.ts:431]: catch error, check if timeout -should be timeout or something?

**Review by LayneHaber — APPROVED** [review]
